Build warnings for `adxl362_decoder.c`/`adxl367_decoder.c`/`adxl372_decoder.c`

### Describe the bug

The drivers for `adi,adxl362`/`adi,adxl367`/`adi,adxl372` perform some pointer juggling, but this results in compiler warnings.

### Regression

- [ ] This is a regression.

### Steps to reproduce

Build the tests for `build_all/sensor`

```shell
$ west build -p -b native_sim/native/64 tests/drivers/build_all/sensor
```

### Relevant log output

```shell
[150/782] Building C object zephyr/drivers/sensor/adi/adxl362/CMakeFiles/drivers__sensor__adi__adxl362.dir/adxl362_decoder.c.obj
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl362/adxl362_decoder.c: In function ‘adxl362_decode_stream’:
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl362/adxl362_decoder.c:92:13: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   92 |         if ((uint8_t *)*fit >= buffer) {
      |             ^
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl362/adxl362_decoder.c:93:31: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   93 |                 sample_num = ((uint8_t *)*fit - buffer) / sample_set_size;
      |                               ^
[164/782] Building C object zephyr/drivers/sensor/adi/adxl367/CMakeFiles/drivers__sensor__adi__adxl367.dir/adxl367_decoder.c.obj
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl367/adxl367_decoder.c: In function ‘adxl367_decode_stream’:
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl367/adxl367_decoder.c:533:21: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  533 |                 if ((uint8_t *)*fit >= buffer) {
      |                     ^
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl367/adxl367_decoder.c:534:39: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  534 |                         sample_num = ((uint8_t *)*fit - buffer) / packet_size;
      |                                       ^
[165/782] Building C object zephyr/drivers/sensor/adi/adxl372/CMakeFiles/drivers__sensor__adi__adxl372.dir/adxl372_decoder.c.obj
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl372/adxl372_decoder.c: In function ‘adxl372_decode_stream’:
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl372/adxl372_decoder.c:61:13: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   61 |         if ((uint8_t *)*fit >= buffer) {
      |             ^
/home/pdgendt/zephyrproject/zephyr/drivers/sensor/adi/adxl372/adxl372_decoder.c:62:31: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
   62 |                 sample_num = ((uint8_t *)*fit - buffer) / sample_set_size;
      |
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: Ubuntu 24.04
- Toolchain: Zephyr SDK 0.16.8
- 869881ef2138e5c69e3dc366114773810cfb114a

### Additional Context

_No response_

--- a/drivers/sensor/adi/adxl362/adxl362_decoder.c
+++ b/drivers/sensor/adi/adxl362/adxl362_decoder.c
@@ -89,8 +89,8 @@ static int adxl362_decode_stream(const uint8_t *buffer, struct sensor_chan_spec 
 	uint64_t period_ns = accel_period_ns[enc_data->accel_odr];
 
 	/* Calculate which sample is decoded. */
-	if ((uint8_t *)*fit >= buffer) {
-		sample_num = ((uint8_t *)*fit - buffer) / sample_set_size;
+	if (*fit >= (uintptr_t)buffer) {
+		sample_num = (*fit - (uintptr_t)buffer) / sample_set_size;
 	}
 
 	while (count < max_count && buffer < buffer_end) {

--- a/drivers/sensor/adi/adxl367/adxl367_decoder.c
+++ b/drivers/sensor/adi/adxl367/adxl367_decoder.c
@@ -530,8 +530,8 @@ static int adxl367_decode_stream(const uint8_t *buffer, struct sensor_chan_spec 
 			data_out, enc_data);
 	} else {
 		/* Calculate which sample is decoded. */
-		if ((uint8_t *)*fit >= buffer) {
-			sample_num = ((uint8_t *)*fit - buffer) / packet_size;
+		if (*fit >= (uintptr_t)buffer) {
+			sample_num = (*fit - (uintptr_t)buffer) / packet_size;
 		}
 
 		while (count < max_count && buffer < buffer_end) {

--- a/drivers/sensor/adi/adxl372/adxl372_decoder.c
+++ b/drivers/sensor/adi/adxl372/adxl372_decoder.c
@@ -58,8 +58,8 @@ static int adxl372_decode_stream(const uint8_t *buffer, struct sensor_chan_spec 
 	uint64_t period_ns = accel_period_ns[enc_data->accel_odr];
 
 	/* Calculate which sample is decoded. */
-	if ((uint8_t *)*fit >= buffer) {
-		sample_num = ((uint8_t *)*fit - buffer) / sample_set_size;
+	if (*fit >= (uintptr_t)buffer) {
+		sample_num = (*fit - (uintptr_t)buffer) / sample_set_size;
 	}
 
 	while (count < max_count && buffer < buffer_end) {

